### ISSUE TITLE
Suggestment 

### ISSUE THREAD

**Issue by Wandstick** [microg/GmsCore#1716]
You should add the huawei sign in option that the vanced microG has added it saves as so much trouble and will actually add so much to us huawei users which can't root their devices to get custom rooms with signature spoofing since OEM unlocking is completely removed from the system.

**Comment by ale5000-git** [comment]
It isn't needed, microG GmsCore since version 0.2.23.214816 allow to spoof device identity.

**Comment by Wandstick** [comment]
YO WHAT can you send the link to download that specific version

**Comment by Wandstick** [comment]
No it does not, it still needs me to have a ROM that supports signature spoofing and that is why we need a huawei sign in button to not confuse new users

**Comment by ale5000-git** [comment]
microG doesn't strictly need signature spoofing to work.
Signature spoofing is needed only to bypass the check of signature spoofing that it is in every app (compiled with Google libraries) but microG alone, including adding an account will still work even without it.
On Huawei it additionally need to spoof another phone but it has nothing to do with signature spoofing.

You can alternatively patch the external app to bypass the signature check and it will still use microG even without signature spoofing.
Vanced didn't do anything special, they have a patched YouTube so obviously they didn't need signature spoofing.

**Comment by ale5000-git** [comment]
Huawei sign-in button was the worst thing ever (an hack at most).
A lot of phones are blocked (all the uncertified phones) so do you expect a separate button for every brand?

Actually on old uncertified phones Google simply say uncertified but still allow it to connect instead new uncertified phones are blocked completely.

**Comment by Wandstick** [comment]
Nope, microg itself doesn't even work without signature spoofing and the proof for that is when I attempt to sign in with a Google account, at the end when I press the "I agree" button it says "Sorry... There was a problem communicating with Google's servers". I mean like if you got a fix for that I would be ultimately thankful for that.

**Comment by Wandstick** [comment]
Also the reason for the huawei sign in being added in the first place was because huawei was the only brand in the brand in the world which cannot be root to therefore patch or install ROM if it could be rooted you wouldn't see me here right now.

**Comment by ale5000-git** [comment]
You are misunderstanding, signature spoofing and device spoofing are separate.
Even if you get root and signature spoofing you will still be unable to add an account.

Signature spoofing is needed to avoid apps saying the Google Play Service isn't installed.
Device spoofing is needed to avoid Google blocking your requests (for example to add an account).

Quote:

> Device profiles
> 
> This features is mostly useful for Huawei devices or other devices that are banned from accessing Google services. Choosing a profile of a non-banned device should allow using the device with Google services.
> 
> For now, two device profiles are included with microG (Nexus 5X with Android 8.1.0 and Moto G with LineageOS 14.1).
> 
> If you are a developer, you can easily create a device profiles based on data gathered from a device. See [this file](https://github.com/microg/GmsCore/blob/master/play-services-core/src/main/res/xml/profile_bullhead_27.xml) for a sample device profile. Once created, you can import device profiles in microG settings (under "Device registration") to test the profile.

**Comment by Wandstick** [comment]
Oh if that's the case, can you link me a download to a supported device profile to the app work on this huawei phone. It has troubled me enough for 2 years without play store.

**Comment by Wandstick** [comment]
Quick question, is it possible for you to help me get microG to work on this huawei phone, since I tried registering with both devices you've mentioned but they had not worked sadly.

**Comment by ale5000-git** [comment]
I don't have huawei myself so I cannot test but it is possible that there are problem with error being cached.

Try to follow the procedure of this user: https://github.com/microg/GmsCore/issues/1651#issuecomment-1034312454

**Comment by ale5000-git** [comment]
Quote:
> * disable wifi/data
> * re-install (or force stop + clear data/cache) microG service core
> * open microG settings, choose profile
> * force stop microG
> * enable wifi/data
> * go to microG settings and disable/enable google cloud registration (make sure it is registered)
> * go login with your account

**Comment by Wandstick** [comment]
OK, I did all of this, I tried going into Google Play and it worked then I pressed sign in and then put my account details and all of that. After it got my account info it brought me back to the sign in page, the account is on the device but the apps just won't display the information got by them. Why is this happening

**Comment by Wandstick** [comment]
Yo, I think I found the solution for huawei users to be able to use microG. So basically I was playing around after enabling device profile and I installed  Gmail, now Gmail allowed you to login with Microsoft yahoo Google etc. I then tried using Microsoft account and it actually worked, so my assumption is that if someone takes microG and just remove the sign in feature, everything will work for use poor huawei users.

**Comment by RJK01-darkside** [comment]
https://github.com/jcchikikomori/GmsCoreHuawei there is a patched microg for huawei

**Comment by ale5000-git** [comment]
RJK01-darkside: Official microG also contain the device spoofing code (to prevent Google ban), others are just outdated versions.

But still there is nothing to do for missing signature spoofing.

**Comment by Wandstick** [comment]
I am surprised no one yet patched play store the same way vanced team did.

**Comment by cheadrian** [comment]
Signature spoofing could be applied Revanced using a custom, per app, [Patches](https://github.com/revanced/revanced-patches).
I'll test a solution for Huawei devices. 
ReVanced Youtube includes a patch for spoofing, for example.

**Comment by RJK01-darkside** [comment]
It is one thing to spoof sign a single Google app versus multiple Google
apps.
John Cyrill Corsanes abandoned his efforts to this project in GitHub .

**Comment by cheadrian** [comment]
It is, but as long as we can't root, unlock bootloader or use other methods to gain access to system files, it's the only way to use GApps without shady Chinese containers or emulators (VMOS, Gspace, etc.).

**Comment by RJK01-darkside** [comment]
Good luck in your endeavors.

**Comment by RJK01-darkside** [comment]
I forgot to mention there is a recently released app lighthouse using 
MicroG with some success. 
https://forum.xda-developers.com/t/so-this-is-a-solution-that-works-for-some-google-apps-i-did-some-test-and-heres-the-result.4515073/

**Comment by cheadrian** [comment]
I saw that, tested it but it doesn't work. I think it is using a Chinese or Asian version of the phone, which still have access to [Huawei MDM API](https://developer.huawei.com/consumer/en/hms/right-sign/).

**Comment by Smoovsky** [comment]
@ale5000-git Man, you are patient.

If I don't get you wrong:
1. Device spoofing is to make Google believe your device is certified which should be easy as it ultimately depends on MicroG to decide what to be sent to Google.
2. Software signature spoofing is for MicroG to deceive other apps which are calling GMS api to believe that MicroG is the authentic Google Play Service and allow themselves to communicate with MicroG. This is much harder, it requires either a modified ROM(i.e. unlocked bootloader) or root access to the device.

What's Revanced team doing is mod those apps and have them trust Revanced MicroG which has not just a different signature but also a different package name.

I'm a software developer but I'm not familiar with Android development at all. I can see Revanced MircoG is currently only serving Youtube and Youtube Music, but I also see they have achieved a lot, especially for Revanced Youtube which includes not just Revanced MiroG mod, but also numerous other mods.

I feel like modding other apps which call GMS api with Revanced tool is not of much difference as modding Youtube, cause I think Youtube itself is already calling 90%, if not all, of GMS account-related apis.

But I don't see many developers out there doing this. I'm not sure whether it's because this is hard, or presumably everyone using MicroG should have access to an unlocked bootload/root privilege hence no need to get around with signature spoofing 
 by modding apps at all.
